packages/cloud_firestore/cloud_firestore/example/android/app/build.gradle

```diff
@@ -1,66 +1,61 @@
 plugins {
     id "com.android.application"
+    // START: FlutterFire Configuration
+    id 'com.google.gms.google-services'
+    // END: FlutterFire Configuration
     id "kotlin-android"
+    // The Flutter Gradle Plugin must be applied after the Android and Kotlin Gradle plugins.
     id "dev.flutter.flutter-gradle-plugin"
-    id "com.google.gms.google-services"
 }
 
 def localProperties = new Properties()
-def localPropertiesFile = rootProject.file('local.properties')
+def localPropertiesFile = rootProject.file("local.properties")
 if (localPropertiesFile.exists()) {
-    localPropertiesFile.withReader('UTF-8') { reader ->
+    localPropertiesFile.withReader("UTF-8") { reader ->
         localProperties.load(reader)
     }
 }
 
-
-def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
+def flutterVersionCode = localProperties.getProperty("flutter.versionCode")
 if (flutterVersionCode == null) {
-    flutterVersionCode = '1'
+    flutterVersionCode = "1"
 }
 
-def flutterVersionName = localProperties.getProperty('flutter.versionName')
+def flutterVersionName = localProperties.getProperty("flutter.versionName")
 if (flutterVersionName == null) {
-    flutterVersionName = '1.0'
+    flutterVersionName = "1.0"
 }
 
-
 android {
-    namespace 'io.flutter.plugins.firebase.firestore.example'
-
-    compileSdk 34
-
-    defaultConfig {
-        applicationId "io.flutter.plugins.firebase.firestore.example"
-        minSdk 21
-        targetSdk 34
-        versionCode flutterVersionCode.toInteger()
-        versionName flutterVersionName
-    }
+    namespace = "io.flutter.plugins.firebase.firestore.example"
+    compileSdk = flutter.compileSdkVersion
```

Comment:
not sure what flutter `compileSdkVersion` is but we will need this to be a minimum of 34 for android BOM >= 33

Reply:
https://github.com/flutter/flutter/blob/124aacbaef3ff0476f3a02c2ab05271e152224db/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy#L45 

it's should we use it or use hardcoded instead ?

Reply:
nice, keep it as it is then 👍